sqlmesh/core/engine_adapter/postgres.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+import logging
+import typing as t
+
+from pandas.io.sql import read_sql_query
+from sqlglot import exp
+
+from sqlmesh.core.engine_adapter.base import EngineAdapterWithIndexSupport
+from sqlmesh.core.engine_adapter.base_postgres import BasePostgresEngineAdapter
+from sqlmesh.core.engine_adapter.shared import TransactionType
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core._typing import TableName
+    from sqlmesh.core.engine_adapter._typing import DF, QueryOrDF
+
+logger = logging.getLogger(__name__)
+
+
+class PostgresEngineAdapter(BasePostgresEngineAdapter, EngineAdapterWithIndexSupport):
```

Comment:
i already made these changes guess we'll conflict

Reply:
i think we can get rid of engine adapterwithindexsupport since you moved index support into a boolean flag, then we don't need base

Reply:
Yep that works I was thinking the same

Reply:
Well we still use the `EngineAdapterWithIndexSupport` here for engines that we haven't fully implemented: https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/engine_adapter/__init__.py#L25-L26